packages/machines/tags-input/src/tags-input.dom.ts

```diff
@@ -14,13 +14,11 @@ export const dom = defineDomHelpers({
   getTagInputId: (ctx: Ctx, opt: TagProps) => ctx.ids?.tagInput?.(opt) ?? `${dom.getTagId(ctx, opt)}:input`,
   getEditInputId: (ctx: Ctx) => `${ctx.editedId}:input`,
 
-  getTagInputEl: (ctx: Ctx, opt: TagProps) =>
-    dom.getRootNode(ctx).getElementById(dom.getTagInputId(ctx, opt)) as HTMLInputElement | null,
-  getRootEl: (ctx: Ctx) => dom.getRootNode(ctx).getElementById(dom.getRootId(ctx)),
-  getInputEl: (ctx: Ctx) => dom.getRootNode(ctx).getElementById(dom.getInputId(ctx)) as HTMLInputElement | null,
-  getHiddenInputEl: (ctx: Ctx) =>
-    dom.getRootNode(ctx).getElementById(dom.getHiddenInputId(ctx)) as HTMLInputElement | null,
-  getEditInputEl: (ctx: Ctx) => dom.getRootNode(ctx).getElementById(dom.getEditInputId(ctx)) as HTMLInputElement | null,
+  getTagInputEl: (ctx: Ctx, opt: TagProps) => dom.getById(ctx, dom.getTagInputId(ctx, opt)) as HTMLInputElement | null,
+  getRootEl: (ctx: Ctx) => dom.getById(ctx, dom.getRootId(ctx)),
+  getInputEl: (ctx: Ctx) => dom.getById(ctx, dom.getInputId(ctx)) as HTMLInputElement | null,
+  getHiddenInputEl: (ctx: Ctx) => dom.getById(ctx, dom.getHiddenInputId(ctx)) as HTMLInputElement | null,
```

Comment:
Can we update getById to take a generic of the type of element and default to `HTMLElement` but the return type should be `T | null`

```js
getById: <T = HTMLElement>(ctx: Ctx, id: string) => dom.getDoc(ctx).getElementById(id) as T | null
```

```suggestion
  getHiddenInputEl: (ctx: Ctx) => dom.getById<HTMLInputElement>(ctx, dom.getHiddenInputId(ctx)),
```